src/chocolatey/infrastructure.app/services/ChocolateyPackageService.cs

```diff
@@ -550,6 +552,11 @@ public void upgrade_noop(ChocolateyConfiguration config)
             return packageUpgrades;
         }
 
+        private void before_package_upgrade(PackageResult packageResult, ChocolateyConfiguration config)
+        {
+            _powershellService.run_package_scripts_for_command(config, packageResult, CommandNameType.upgrade);
+        }
+

```

Comment:
Ah... but in the uninstall scenario both scripts are run from the same package version. In the upgrade scenario one script comes from the old package, and one from the new.

As currently implemented in master/stable, an "upgrade" appears to be converted into what PowershellService sees as an "uninstall" followed by an install of the new version.

Ah... I think I now see what you are worried about though. my current implementation (which I suspect is non functional - I think I goofed somewhere during a rebase) may be running beforeModify twice. I'll check that.